Features/ErrorViews/ErrorViewController.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  ErrorViewController.swift
+//  Movies
+//
+//  Created by Diana Chizhik on 18.07.22.
+//
+
+import UIKit
+
+
+class ErrorViewController: UIViewController, ErrorViewDelegate {
+    private var error: MovieServiceError
+    
+    lazy var contentView: ErrorView = {
+        let view = ErrorView()
+        view.delegate = self
+        return view
+    }()
+    
+    init(error: MovieServiceError) {
+        self.error = error
+        
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    @available(*, unavailable)
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func loadView() {
+        self.view = contentView
+        
+        switch error {
+        case .failedToGetResponse:
+            contentView.imageView.image = UIImage(named: "wifi")
+            contentView.errorLabel.text = "Sherlock didn’t find the internet signal.\nPlease try again later."
+        case .failedToGetData:
+            contentView.imageView.image = UIImage(named: "dizzy")
+            contentView.errorLabel.text = "Houston, we have a problem.\nClose and re-open the app."
+        case .failedToDecode:
+            contentView.imageView.image = UIImage(named: "spy")
+            contentView.errorLabel.text = "Failed to decode data.\nCall the special agents."
+        }
```

Comment:
With this approach, on every new error added we would have to manually come here and add the proper handling. Do you see any different approach? Perhaps a way to handle any current or future errors.
Bonus: Can we handle any kind of error, even those who are not `MovieServiceError`?

Reply:
I'll extend MovieServiceError enum with image property.

We can since MovieServiceError conforms to Error protocol. So, I'll use Error protocol as type and pass it as parameter type to ErrorVC init. Then downcast to handle known errors.